File: troc-12/troc-12-sosial/solution_rama.py
# Author   : rama_pang
# Problem  : troc-12-sosial
# Time     : O(N)
# Memory   : O(N)

from sys import stdin, stdout

def main():
  N = int(stdin.readline());
  A = [int(x) for x in stdin.readline().split()]
  S = 0
  for i in range(N + 1):
    S += A[i] * (2 ** i)
  
  if S % (2 ** N) == 0:
    stdout.write(str(S // (2 ** N)) + "\n")
  else:
    stdout.write("-1\n")

if __name__ == '__main__':
  main()

File: troc-12/troc-12-sosial/solution_rama.cpp
// Author   : rama_pang
// Problem  : troc-12-sosial
// Time     : O(N)
// Memory   : O(1)

#include <bits/stdc++.h>
using namespace std;

int main() {
  int N;
  cin >> N;

  long long sum = 0;
  for (int i = 0; i <= N; i++) {
    int A;
    cin >> A;
    sum += (1ll << i) * A;
  }
  
  cout << (sum % (1 << N) == 0 ? sum / (1 << N) : -1) << "\n";
  return 0;
}
